operators/src/source/ogr_source.rs

```diff
@@ -863,15 +914,29 @@ where
         was_time_filtered_by_ogr: bool,
         was_spatial_filtered_by_ogr: bool,
     ) -> Result<()> {
+        let error_spec = dataset_information.on_error;
+
         let time_interval = time_extractor(feature)?;
 
         // filter out data items not in the query time interval
         if !was_time_filtered_by_ogr && !time_interval.intersects(&query_rectangle.time_interval) {
             return Ok(());
         }
 
-        let geometry: G =
-            <G as TryFromOgrGeometry>::try_from(feature.geometry_by_index(0).map_err(Into::into))?;
+        let geometry: G = match <G as TryFromOgrGeometry>::try_from(
+            feature.geometry_by_index(0).map_err(Into::into),
+        ) {
+            Ok(g) => g,
+            Err(Error::Gdal {
+                source: GdalError::InvalidFieldIndex { .. },
+            }) if dataset_information.default_geometry.is_some() => dataset_information
+                .default_geometry
+                .as_ref()
+                .expect("Impossible")
+                .clone()
+                .try_into()?,
```

Comment:
You could try to get the right geometry from the method before and insert an Option<G> here.

Maybe this will get the method more concise. Do you have any idea if there would be a good way to get rid of the *if_some() then unwrap()* pattern? If not, we can keep it as is.

Reply:
023113f1a5d7d0dd935dc67b36f3d1ddd322ce5b